File: _0_language/_concurrency/_2_thread_lifecycle_and_management/_threads_lifecycle/threads_lifecycle.py
import threading
import time

def print_message():
    print("Thread is running!")
    time.sleep(2)
    print("Thread has finished!")

thread = threading.Thread(target=print_message)
print("State after creation:", thread.is_alive())

thread.start()  # Start the thread
print("State after start:", thread.is_alive())

thread.join()  # Wait for the thread to finish
print("State after termination:", thread.is_alive())

File: _0_language/_concurrency/_2_thread_lifecycle_and_management/_threads_lifecycle/threads_lifecycle.go
package main

import "fmt"
import "time"

func printMessage() {
    fmt.Println("Thread is running!")
    time.Sleep(2 * time.Second)
    fmt.Println("Thread has finished!")
}

func main() {
    fmt.Println("Before goroutine")
    go printMessage()  // Start the goroutine
    time.Sleep(3 * time.Second)  // Wait for goroutine to finish
    fmt.Println("After goroutine")
}
